### PR TITLE
Fixes #284 - Loops/comprehensions over decreasing ranges don't work

### LINKED ISSUE #284: Loops/comprehensions over decreasing ranges don't work
```
$ coffee -be --cli "console.log (n for n in [5..-5])"
[]
$ coffee -bj --cli "console.log (n for n in [5..-5])"
// Generated by CoffeeScript 2.0.0-beta9-dev
var n;
console.log(function (accum$) {
  for (i$ = 5; i$ <= -5; ++i$) {
    n = i$;
    accum$.push(n);
  }
  return accum$;
}.call(this, []));
```

jashkenas/coffee-script produces the following:

```
$ coffee -be "console.log (n for n in [5..-5])"
[ 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5 ]
$ coffee -bce "console.log (n for n in [5..-5])"
// Generated by CoffeeScript 1.7.1
var n;

console.log((function() {
  var _i, _results;
  _results = [];
  for (n = _i = 5; _i >= -5; n = --_i) {
    _results.push(n);
  }
  return _results;
})());
```
## 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/2268299-loops-comprehensions-over-decreasing-ranges-don-t-work?utm_campaign=plugin&utm_content=tracker%2F33145&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F33145&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by basicer**

### CODE REVIEW

File: src/compiler.coffee [hunk 1]
```diff
@@ -316,6 +316,21 @@ generateSoak = do ->
     [tests, e] = fn node
     new CS.Conditional (foldl1 tests, (memo, t) -> new CS.LogicalAndOp memo, t), e
 
+extractNumber = (what) ->
+  return what.data if what.instanceof CS.Int
+  return false unless what.instanceof CS.UnaryNegateOp
+  return false unless what.expression.instanceof CS.Int
+  return 0-what.expression.data
```
Comment (michaelficarra) [inline src/compiler.coffee]: Space infix operators.

```coffee
0 - what.expression.data
```

File: src/compiler.coffee [hunk 2]
```diff
@@ -316,6 +316,21 @@ generateSoak = do ->
...
+
+extractStaticRange = (range) ->
+  return undefined unless range.instanceof CS.Range
+  left = extractNumber(range.left)
```
Comment (michaelficarra) [inline src/compiler.coffee]: Drop parens for function application.

```coffee
extractNumber range.left
```

File: src/compiler.coffee [hunk 3]
```diff
@@ -316,6 +316,21 @@ generateSoak = do ->
...
+  right = extractNumber(range.right)
+
+  return undefined if left == false
+  return undefined if right == false
+
+  return [left,right]
```
Comment (michaelficarra) [inline src/compiler.coffee]: Space array elements.

```coffee
[left, right]
```

**Comment by basicer** [comment]
@michaelficarra Updated 😄

**Pull request merged**